bool found
string image_name
